server-kotlin/src/main/kotlin/digital/cesko/full_text_search/service/SearchService.kt

```diff
@@ -0,0 +1,64 @@
+package digital.cesko.full_text_search.service
+
+import digital.cesko.full_text_search.model.SearchResult
+import org.apache.lucene.document.IntPoint
+import org.apache.lucene.index.DirectoryReader
+import org.apache.lucene.index.Term
+import org.apache.lucene.search.BooleanClause
+import org.apache.lucene.search.BooleanQuery
+import org.apache.lucene.search.IndexSearcher
+import org.apache.lucene.search.WildcardQuery
+import org.apache.lucene.store.Directory
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.annotation.Scope
+import org.springframework.stereotype.Service
+
+
+const val CONTRACT_TITLE = "title"
+const val CONTRACT_COUNTERPARTY = "counterparty"
+const val CONTRACT_AMOUNT = "amount"
+const val CONTRACT_CURRENCY = "currency"
+const val CONTRACT_URL = "url"
+const val CONTRACT_PROFILE = "profile"
+const val CONTRACT_DATE = "date"
+const val CONTRACT_ID = "id"
+const val MAX_SIZE = 30
+
+@Service
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+class SearchService(
+        private var directory: Directory
+) {
+    fun countById(contractId: Int): Int {
+        val searcher = IndexSearcher(DirectoryReader.open(directory))
+        return searcher.count(IntPoint.newExactQuery(CONTRACT_ID, contractId))
+    }
+
+    fun search(stringQuery: String, profile: String?): List<SearchResult> {
+        val contractCounterParty = WildcardQuery(Term(CONTRACT_COUNTERPARTY, "$stringQuery*".toLowerCase()))
+        val contractTitle = WildcardQuery(Term(CONTRACT_TITLE, "$stringQuery*".toLowerCase()))
```

Comment:
Co funguje dobre je hledat na dvakrat:
- Prvni vyhledavani hleda vsechny slova presne a posledni slovo s * (radar br => radar AND br*)
- Kdyz prvni nic nenajde (0 vysledku) prijde druhe kolo s povolenim chyb (radar br => radar~ AND br~)

Tohle dobre odpovida tomu, co clovek chce behem psani => prvni slova, co napsal, sou kompletni a posledni se doplnuje a kdyz se nic nenajde, tak zkusi jestli v nejakem slove neni chyba.

Reply:
ok, obslehl jsem to z city searche